pyschema/source_generation.py

```diff
@@ -78,48 +194,50 @@ def _class_source(schema, indent):
 
 class CachedGraphTraverser(object):
     def __init__(self):
-        self.descendents = {}
+        self.descendants = {}
         self.started = set()
 
-    def find_descendents(self, a):
-        if a in self.descendents:
+    def find_descendants(self, a, max_depth=sys.getrecursionlimit()):
+        if a in self.descendants:
             # fetch from cache
-            return self.descendents[a]
+            return self.descendants[a]
         self.started.add(a)
         subs = set()
-        if pyschema.ispyschema(a):
-            for _, field in a._fields.iteritems():
-                subs |= self.find_descendents(field)
-            self.descendents[a] = subs
-        elif isinstance(a, types.List):
-            subs |= self.find_descendents(a.field_type)
-        elif isinstance(a, types.Map):
-            subs |= self.find_descendents(a.value_type)
-        elif isinstance(a, types.SubRecord):
-            subs.add(a._schema)
-            if a not in self.started:  # otherwise there is a circular reference
-                subs |= self.find_descendents(a._schema)
-        self.started.remove(a)
+        if max_depth > 0:
```

Comment:
I'm conflicted what is cleaner: having a big indented block, or inverting the if and returning the empty set early in the function to keep indentation at its current level. You decide :)